agent/statemanager/state_manager_test.go

```diff
@@ -486,3 +487,43 @@ func TestLoadsDataSeqTaskManifest(t *testing.T) {
 	assert.EqualValues(t, 0, sequenceNumber)
 	assert.EqualValues(t, 7, seqNumTaskManifest)
 }
+
+func TestLoadsDataForGMSATask(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skip()
+	}
+	cleanup, err := setupWindowsTest(filepath.Join(".", "testdata", "v26", "gmsa", "ecs_agent_data.json"))
+	require.Nil(t, err, "Failed to set up test")
+	defer cleanup()
+	cfg := &config.Config{DataDir: filepath.Join(".", "testdata", "v26", "gmsa")}
+	taskEngine := engine.NewTaskEngine(&config.Config{}, nil, nil, nil, nil, dockerstate.NewTaskEngineState(), nil, nil)
+	var containerInstanceArn, cluster, savedInstanceID string
+	var sequenceNumber int64
+	stateManager, err := statemanager.NewStateManager(cfg,
+		statemanager.AddSaveable("TaskEngine", taskEngine),
+		statemanager.AddSaveable("ContainerInstanceArn", &containerInstanceArn),
+		statemanager.AddSaveable("Cluster", &cluster),
+		statemanager.AddSaveable("EC2InstanceID", &savedInstanceID),
+		statemanager.AddSaveable("SeqNum", &sequenceNumber),
+	)
+	assert.NoError(t, err)
+	err = stateManager.Load()
+	assert.NoError(t, err)
+	assert.Equal(t, "gmsa-test", cluster)
+	assert.EqualValues(t, 0, sequenceNumber)
+	tasks, err := taskEngine.ListTasks()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(tasks))
+	task := tasks[0]
+
+	assert.Equal(t, "arn:aws:ecs:ap-northeast-1:1234567890:task/b8e2bd3c-c82a-4b43-9bde-199ae05b49a5", task.Arn)
+	assert.Equal(t, "gmsa-test", task.Family)
+	assert.Equal(t, 1, len(task.Containers))
+	container := task.Containers[0]
+	assert.Equal(t, "windows_sample_app", container.Name)
+	assert.NotNil(t, container.DockerConfig.HostConfig)
+
+	resource, ok := task.GetCredentialSpecResource()
+	assert.True(t, ok)
+	assert.NotEmpty(t, resource)
```

Comment:
Can you also check the attributes of the task resource? You need to depend on the state file value when agent restarts in order for container created correctly. So you need make sure all attributes you depend on have correct values there.

Reply:
That would make the test depend on the platform as the CredSpecMap is only available on windows. The test would fail on other platforms.

Reply:
you have the check that this test will not be run on platforms other than windows.

Reply:
It will still result in a failure on other platforms when you build a test binary for execution.

Reply:
This cannot be the reason that we don't check the value here. As a temporary workaround, considering windows tests fix needs some time, you can add some essential attributes/functions in credentialspec_unsupported.go file to make it work. Otherwise, you will need to refactor the windows statemanager test or wait for us to fix it.

Reply:
Changing the functional code to accommodate for test infrastructure issues is never a good idea.

Reply:
I have updated the test.